packages/gallery/src/components/layoutFixer/createLayoutFixer.js

```diff
@@ -98,63 +99,72 @@ export const createLayoutFixer = () => {
                 this.items = JSON.parse(this.getAttribute('items')).map(convertDtoToLayoutItem);
                 if (!(this.items && this.items.length > 0)) {
                     this.useLayouter = false
-                } else {
-                    window.layoutFixer[this.domId].items = this.items;
                 }
 
                 this.styleParams = JSON.parse(this.getAttribute('styles'));
-                
+
                 if (!(this.styleParams && typeof this.styleParams === 'object')) {
                     this.useLayouter = false
-                } else {
-                    window.layoutFixer[this.domId].styles = this.styleParams;
                 }
 
-                const {width, height, top} = this.parent && this.parent.getBoundingClientRect();
+                const { width, height, top } = this.parent && this.parent.getBoundingClientRect();
                 this.measures = {
                     top,
                     width,
                     height,
                     scrollBase: top,
                 };
-                window.layoutFixer[this.domId].container = this.measures;
                 console.log('[LAYOUT FIXER] measured container', this.measures);
-               
-                if (this.measures && this.useLayouter && typeof createLayout === 'function') {
-                    this.layout = createLayout({
+
+                if (this.measures && this.useLayouter) {
+                    this.blueprintsManager = new BlueprintsManager({ id: 'layoutFixer' });
+                    this.blueprintsManager.createBlueprint({
                         items: this.items,
                         styleParams: this.styleParams,
                         container: this.measures
-                    })
-                    console.log('[LAYOUT FIXER] created layout', this.layout);
-                    window.layoutFixer[this.domId].structure = this.layout;
-                }
+                    }).then(blueprint => {
+                        console.log('[LAYOUT FIXER] created blueprint', blueprint);
+                        Object.assign(window.layoutFixer[this.domId], blueprint);
+                        this.layout = blueprint.structure;
```

Comment:
Why do we need this? we can save the blueprint on 'this' and use this.blueprints.structure instead of having a new param called layout.